### ISSUE TITLE
Feature: support for downloading analytics data from ITC

### ISSUE THREAD

**Issue by lacostej** [fastlane/fastlane#3682]
### Issue Description

I started reviving fastlane/spaceship#190

The original PR allowed to fetch time_series, and I noticed that ITC analytics contain lots of data (Overview, Metrics, Sources, Retention). The first thing I will do is to check what people want !

So far this is what I found so far. Here are the URLs that are hit to fetch data, depending on the main views:

##### Overview
* "data/app/detail/all-time"
* "data/app/detail/measures"

##### Metrics
* "data/app/detail/dimensions"
* "data/app/detail/retention"
* "data/app/detail/measures"
* "userpreferences/METRICSVIEW"
* "data/events"
* "data/dimension-values"
* "data/time-series"

##### Sources
* "data/sources/campaign-list"
* "data/sources/domainreferrer-list"

##### Retention
* "data/retention"

CC @pieisgood

**Comment by pieisgood** [comment]
Essentially, I'm only after time series data. But having access to the rest of the data isn't a bad thing.

**Comment by lacostej** [comment]
@pieisgood are you interested in trying out the current API ?

Maybe give some feedback. I have still not decided the exact API style

See https://github.com/lacostej/fastlane/commit/analytics

**Comment by shulard** [comment]
Hello !It's actually related to ITC only ? Do you think it can also be done for the play store ?

**Comment by pieisgood** [comment]
Spaceship is related to iTunes Connect only, that's correct. I'm not sure how the expansion of fastlane into Android is planned out, but I thought there were already APIs for google play store analytics data?

**Comment by shulard** [comment]
I can't find that kind of API on Google Play store... There are some mention of Google Cloud CSV data that can be retrieved but nothing really solid... I'm gonna continue searching about, thanks for your answer (and your awesome work on that issue :+1:).

**Comment by pieisgood** [comment]
Good luck! If you find anything post it here since I'll be in the same boat soon enough.

**Comment by ravinggenius** [comment]
I'm looking for a way to get daily download numbers. Can I help with anything?

**Comment by pieisgood** [comment]
At this point, you can either build on top of what I built and make another pull request or do the same thing for what lacostej built. My code still works and I use it daily, so you can just sorta ham fist it into the spaceship section of your fastlane ruby gem on your disk... if you don't want to make a pull request.

**Comment by kumarSC** [comment]
Hey @pieisgood , I was trying to use iTunes api from your fork, https://github.com/pieisgood/spaceship but getting an error while running it from command line.

```
[5] pry(main)> app.analytics.app_views()
=> {"itcBaseUrl"=>"https://itunesconnect.apple.com",
 "message"=>"Make sure the URL is correct and try again.",
 "l10n"=>{"Crossfire-response-badrequest-title"=>"The URL is invalid.", "Crossfire-response-badrequest-message"=>"Make sure the URL is correct and try again."},
 "errors"=>[{"message"=>"Invalid adam id: null", "messageTemplate"=>"Invalid adam id: null", "invalidValue"=>"[null]"}]}
```
Appreciate any help.

**Comment by pieisgood** [comment]
Did you initialize the app? the adam id is null... so no app was chosen to
get analytics for it seems
